File: cidr_convert_api/go/convert_test.go
package main

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestValidCidrToMask(t *testing.T) {
	assert := assert.New(t)

	assert.Equal("128.0.0.0", cidrToMask("1"))
	assert.Equal("255.255.0.0", cidrToMask("16"))
	assert.Equal("255.255.248.0", cidrToMask("21"))
	assert.Equal("255.255.255.255", cidrToMask("32"))
}

func TestValidMaskToCidr(t *testing.T) {
	assert := assert.New(t)

	assert.Equal("1", maskToCidr("128.0.0.0"))
	assert.Equal("16", maskToCidr("255.255.0.0"))
	assert.Equal("21", maskToCidr("255.255.248.0"))
	assert.Equal("32", maskToCidr("255.255.255.255"))
}

func TestValidIpv4(t *testing.T) {
	assert := assert.New(t)

	assert.Equal(true, ipv4Validation("127.0.0.1"))
	assert.Equal(true, ipv4Validation("0.0.0.0"))
	assert.Equal(true, ipv4Validation("192.168.0.1"))
	assert.Equal(true, ipv4Validation("255.255.255.255"))
}


func TestInvalidMaskToCidr(t *testing.T) {
	assert := assert.New(t)

	assert.Equal("Invalid", maskToCidr("0.0.0.0"))
	assert.Equal("Invalid", maskToCidr("-1"))
	assert.Equal("Invalid", maskToCidr("33"))
}

func TestInvalidCidrToMask(t *testing.T) {
	assert := assert.New(t)

	assert.Equal("Invalid", cidrToMask("0"))
	assert.Equal("Invalid", cidrToMask("0.0.0.0.0"))
	assert.Equal("Invalid", cidrToMask("255.255.255"))
	assert.Equal("Invalid", cidrToMask("11.0.0.0"))
}


func TestInvalidIpv4(t *testing.T) {
	assert := assert.New(t)

	assert.Equal(false, ipv4Validation("192.168.1.2.3"))
	assert.Equal(false, ipv4Validation("a.b.c.d"))
	assert.Equal(false, ipv4Validation("255.256.250.0"))
	assert.Equal(false, ipv4Validation("...."))

}

File: cidr_convert_api/java/cidr-api/src/test/java/com/wizeline/ConvertTest.java
package com.wizeline;
import static com.wizeline.Convert.*;
import junit.framework.Test;
import junit.framework.TestCase;
import junit.framework.TestSuite;

/**
 * Unit test for simple App.
 */
public class ConvertTest extends TestCase
{
    /**
     * Create the test case
     *
     * @param testConvertClass name of the test case
     */
    public ConvertTest( String testName )
    {
        super( testName );
    }

    /**
     * @return the suite of tests being tested
     */
    public static Test suite()
    {
        return new TestSuite( ConvertTest.class );
    }

    public void testValidCidrToMask()
    {
      assertEquals("128.0.0.0", Convert.cidrToMask("1"));
      assertEquals("255.255.0.0", Convert.cidrToMask("16"));
      assertEquals("255.255.248.0", Convert.cidrToMask("21"));
      assertEquals("255.255.255.255", Convert.cidrToMask("32"));
    }

    public void testValidMaskToCidr()
    {
      assertEquals("1", Convert.maskToCidr("128.0.0.0"));
      assertEquals("16", Convert.maskToCidr("255.255.0.0"));
      assertEquals("21", Convert.maskToCidr("255.255.248.0"));
      assertEquals("32", Convert.maskToCidr("255.255.255.255"));
    }

    public void testValidIpv4()
    {
      assertTrue(Convert.ipv4Validation("127.0.0.1"));
      assertTrue(Convert.ipv4Validation("0.0.0.0"));
      assertTrue(Convert.ipv4Validation("192.168.0.1"));
      assertTrue(Convert.ipv4Validation("255.255.255.255"));
    }

    public void testInvalidMaskToCidr()
    {
      assertEquals("Invalid", Convert.maskToCidr("0.0.0.0"));
      assertEquals("Invalid", Convert.maskToCidr("-1"));
      assertEquals("Invalid", Convert.maskToCidr("33"));
    }

    public void testInvalidCidrToMask()
    {
      assertEquals("Invalid", Convert.cidrToMask("0"));
      assertEquals("Invalid", Convert.cidrToMask("0.0.0.0.0"));
      assertEquals("Invalid", Convert.cidrToMask("255.255.255"));
      assertEquals("Invalid", Convert.cidrToMask("11.0.0.0"));
    }

    public void testInvalidIpv4()
    {
      assertFalse(Convert.ipv4Validation("192.168.1.2.3"));
      assertFalse(Convert.ipv4Validation("a.b.c.d"));
      assertFalse(Convert.ipv4Validation("255.256.250.0"));
      assertFalse(Convert.ipv4Validation("...."));
    }
}
